pkg/proof/proof.go

```diff
@@ -51,22 +51,21 @@ func getTxNamespace(tx []byte) (ns appns.Namespace) {
 	return appns.TxNamespace
 }
 
-// NewShareInclusionProof returns an NMT inclusion proof for a set of shares to the data root.
+// NewShareInclusionProof returns an NMT inclusion proof for a set of shares
+// belonging to the same namespace to the data root.
 // Expects the share range to be pre-validated.
-// Note: only supports inclusion proofs for shares belonging to the same namespace.
 func NewShareInclusionProof(
-	allRawShares []shares.Share,
-	squareSize uint64,
+	dataSquare square.Square,
 	namespace appns.Namespace,
-	startShare uint64,
-	endShare uint64,
+	shareRange shares.Range,
 ) (types.ShareProof, error) {
-	startRow := startShare / squareSize
-	endRow := endShare / squareSize
-	startLeaf := startShare % squareSize
-	endLeaf := endShare % squareSize
+	squareSize := dataSquare.Size()
+	startRow := shareRange.Start / squareSize
+	endRow := (shareRange.End - 1) / squareSize
```

Comment:
why `-1` ? the NMT takes end exclusive ranges too, shouldn't this be `shareRange.End`?

Reply:
Because the endRow might be incorrect if we are using the index of the share that actually comes after the last share

Reply:
So we expect Core to send the ranges already end exclusive?

Reply:
Are you referring to the `ProveShares` rpc endpoint? Core just forwards whatever the users request is, so it will need to be the user that updates and specifies an end exclusive share range.

We should update documentation and comments if we can though